tools/test_tool_option_parsing.py

```diff
@@ -0,0 +1,317 @@
+    args = rmlvos + ['0x1234']
+    xkbcli_how_to_type.run_command_success(args)
+
+
+@pytest.mark.parametrize('args', [['--verbose'],
+                                  ['-v'],
+                                  ['--verbose', '--load-exotic'],
+                                  ['--load-exotic'],
+                                  ['--ruleset=evdev'],
+                                  ['--ruleset=base'],
+                                  ])
+def test_list_rmlvo(xkbcli_list, args):
+    xkbcli_list.run_command_success(args)
+
+
+def test_list_rmlvo_includes(xkbcli_list):
+    args = ['/tmp/']
+    xkbcli_list.run_command_success(args)
+
+
+def test_list_rmlvo_includes_invalid(xkbcli_list):
+    args = ['/proc/version']
+    rc, stdout, stderr = xkbcli_list.run_command(args)
+    assert rc == 1
+    assert "Failed to append include path" in stderr
+
+
+def test_list_rmlvo_includes_no_defaults(xkbcli_list):
+    args = ['--skip-default-paths', '/tmp']
+    rc, stdout, stderr = xkbcli_list.run_command(args)
+    assert rc == 1
+    assert "Failed to parse XKB description" in stderr
+
+
+@pytest.mark.skipif(not os.path.exists('/dev/input/event0'), reason='event node required')
+@pytest.mark.skipif(not os.access('/dev/input/event0', os.R_OK), reason='insufficient permissions')
+@pytest.mark.parametrize('rmlvos', rmlvos)
+def test_interactive_evdev_rmlvo(xkbcli_interactive_evdev, rmlvos):
+    return
+    xkbcli_interactive_evdev.run_command_success(rmlvos)
+
+
+@pytest.mark.skipif(not os.path.exists('/dev/input/event0'),
+                    reason='event node required')
+@pytest.mark.skipif(not os.access('/dev/input/event0', os.R_OK),
+                    reason='insufficient permissions')
+@pytest.mark.parametrize('args', [['--report-state-changes'],
+                                  ['--enable-compose'],
+                                  ['--consumed-mode=xkb'],
+                                  ['--consumed-mode=gtk'],
+                                  ])
+def test_interactive_evdev(xkbcli_interactive_evdev, args):
+    # Note: --enable-compose fails if $prefix doesn't have the compose tables
+    # installed
+    xkbcli_interactive_evdev.run_command_success(args)
+
+
+@pytest.mark.skipif(not os.getenv('DISPLAY'), reason='DISPLAY not set')
+def test_interactive_x11(xkbcli_interactive_x11):
+    # To be filled in if we handle something other than --help
+    pass
+
+
+@pytest.mark.skipif(not os.getenv('WAYLAND_DISPLAY'),
+                    reason='WAYLAND_DISPLAY not set')
+def test_interactive_wayland(xkbcli_interactive_wayland):
+    # To be filled in if we handle something other than --help
+    pass
+
+
+def main():
+    args = ['-m', 'pytest']
+    try:
+        import xdist  # noqa
+        args += ['-n', 'auto']
+    except ImportError:
+        logger.info('python-xdist missing, this test will be slow')
+        pass
+
+    args += ['@MESON_BUILD_ROOT@']
+
+    return subprocess.run([sys.executable] + args).returncode
```

Comment:
Can run `pytest` in-process using [`pytest.main()`](https://docs.pytest.org/en/stable/reference.html#pytest-main), if you want.

Reply:
worked, but didn't :) Problem is that it went off to search for test files and found the source file in `tools` which always fails. The trick is to pass `@MESON_BUILD_ROOT@` in as args but it also made me realise that we can avoid this whole situation by renaming the source file to `test-tool-option-parsing.py` so pytest doesn't recognise it. This also removes the need for the warning.